ngraph/python/tests/runtime.py

```diff
@@ -0,0 +1,112 @@
+
+    def __init__(self, backend_name: str) -> None:
+        self.backend_name = backend_name
+        # Plugin initialization for specified device and load extensions library if specified
+        log.info("Creating Inference Engine")
+        ie = IECore()
+        ie.set_config({}, device_name=backend_name)
+        self.backend = ie
+
+    def set_config(self, config: Dict[str, str]) -> None:
+        """Set the inference engine configuration."""
+        self.backend.set_config(config, device_name=self.backend_name)
+
+    def __repr__(self) -> str:
+        return "<Runtime: Backend='{}'>".format(self.backend_name)
+
+    def computation(self, node_or_function: Union[Node, Function], *inputs: Node) -> "Computation":
+        """Return a callable Computation object."""
+        if isinstance(node_or_function, Node):
+            ng_function = Function(node_or_function, inputs, node_or_function.name)
+            return Computation(self, ng_function)
+        elif isinstance(node_or_function, Function):
+            return Computation(self, node_or_function)
+        else:
+            raise TypeError(
+                "Runtime.computation must be called with an nGraph Function object "
+                "or an nGraph node object an optionally Parameter node objects. "
+                "Called with: %s",
+                node_or_function,
+            )
+
+
+class Computation(object):
+    """nGraph callable computation object."""
+
+    def __init__(self, runtime: Runtime, ng_function: Function) -> None:
+        ie = runtime.backend
+        self.runtime = runtime
+        self.function = ng_function
+        self.parameters = ng_function.get_parameters()
+        self.results = ng_function.get_results()
+
+        capsule = Function.to_capsule(ng_function)
+        cnnNetwork = IENetwork(capsule)
+        self.executable_network = ie.load_network(cnnNetwork, 'CPU')
+
+    def __repr__(self) -> str:
+        params_string = ", ".join([param.name for param in self.parameters])
+        return "<Computation: {}({})>".format(self.function.get_name(), params_string)
+
+    def __call__(self, *input_values: NumericData) -> List[NumericData]:
+        """Run computation on input values and return result."""
+
+        # Input validation
+        if len(input_values) != len(self.parameters):
+            raise UserInputError(
+                "Expected %s parameters, received %s.", len(self.parameters), len(input_values)
+            )
+        for parameter, input in zip(self.parameters, input_values):
+            if list(parameter.shape) != list(input.shape) and len(input.shape) > 0:
+                raise UserInputError(
+                    "Provided tensor's shape: %s does not match the expected: %s.",
+                    list(input.shape),
+                    list(parameter.shape),
+                )
+
+        request = self.executable_network.requests[0]
+        request.infer(dict(zip(request._inputs_list, input_values)))
```

Comment:
Are you sure that we may zip input with values in order? Doesn't IE distinguish inputs as well outputs by name rather than by index?